### DISCUSSION TITLE
New game mod - Numpad Hotkeys

### DISCUSSION THREAD

**Discussion by 0xdeadbeer** [ppy/osu#16329] [Ideas]
This one is actually an idea I came up with while watching a tokaku's video and the idea does not even sound that wrong to me (I'm surprised it wasn't added yet in osu! lazer). 
The whole idea is that you use your numpad as hotkeys to play. Say a circle is spawning with the number 8 on it and you will have to click the num 8 instead of the button C/V/whatever your hotkey for left/right click on osu! is. And for the keyboards that don't have a numpad there will be a mod preference/option to change the hotkeys to the alphabet characters for example..

Inspiration source: https://pulsus.cc/play/

What do you guys think about this one?

**Comment by bdach** [comment]
what happens when the number is higher than 9?

**Reply by 0xdeadbeer** [reply]
now here i wonder what's the max number a circle can have? if its just 15 or smth that should be doable (add more keys or add combos of keys).. on the other hand if the max number is smth like 99.. well.. we gotta find another way to sort them by number.. either by the colour or something else idk. Will think about it and report once i finish implementing a test version of it onto my own osu repo.

**Reply by bdach** [reply]
there is no max number, theoretically. other than the bounds of 32 bit integers maybe.

**Reply by 0xdeadbeer** [reply]
> there is no max number, theoretically. other than the bounds of 32 bit integers maybe.

I see, consider though that It should (im supposing) never get to a point where it will display numbers higher than 99.. as 2 digits already barely fit inside a circle... 3 digits could never fit inside. I'll see what to do about it, myb group them by colours or something else.. ✌️

**Reply by bdach** [reply]
au contraire: https://osu.ppy.sh/beatmapsets/595198#osu/1258704

![image](https://user-images.githubusercontent.com/20418176/148302795-f03d2fa4-9b2f-40e9-88b4-a754922e21e6.png)

**Reply by Theighlin** [reply]
What is the effect of this type of combo on the game? If there is none maybe the mod could just split higher combo into smaller ones on the fly?

**Reply by TheBrunoRM** [reply]
What about doing the modulus operation of the combo?
The operation would be like this: (combo-1%9)+1
e.g. combo 15 would be 6

**Reply by 0xdeadbeer** [reply]
> What about doing the modulus operation of the combo?
> The operation would be like this: (combo-1%9)+1
> e.g. combo 15 would be 6

Mhm! yeah that could solve the problem

**Reply by Semro** [reply]
@osamu-kj to clarify, do you mean that a player has to aim circles while with another hand figuring out what a key to hit?

**Reply by 0xdeadbeer** [reply]
> @osamu-kj to clarify, do you mean that a player has to aim circles while with another hand figuring out what a key to hit?

Yup! u got it
